internal/pkg/dl/agent.go

```diff
@@ -48,19 +49,26 @@ func prepareOfflineAgentsByPolicyID() *dsl.Tmpl {
 	return tmpl
 }
 
-func FindAgent(ctx context.Context, bulker bulk.Bulk, tmpl *dsl.Tmpl, name string, v interface{}, opt ...Option) (agent model.Agent, err error) {
+func FindAgent(ctx context.Context, bulker bulk.Bulk, tmpl *dsl.Tmpl, name string, v interface{}, opt ...Option) (model.Agent, error) {
 	o := newOption(FleetAgents, opt...)
 	res, err := SearchWithOneParam(ctx, bulker, tmpl, o.indexName, name, v)
 	if err != nil {
-		return
+		return model.Agent{}, fmt.Errorf("failed searching for agent: %w", err)
 	}
 
 	if len(res.Hits) == 0 {
-		return agent, ErrNotFound
+		return model.Agent{}, ErrNotFound
 	}
 
-	err = res.Hits[0].Unmarshal(&agent)
-	return agent, err
+	var agent model.Agent
+	if err = res.Hits[0].Unmarshal(&agent); err != nil {
+		return model.Agent{}, fmt.Errorf("could not unmarshal ES document into model.Agent: %w", err)
+	}
+
+	if agent.ElasticsearchOutputs == nil {
+		agent.ElasticsearchOutputs = map[string]*model.PolicyOutput{}
+	}
```

Comment:
Does this have to be here or could the other parts of the code just check for `nil`? This is allocating memory when it seems like a better idea would be to not allocate any.

Reply:
not really. I just put here to make it safer/easier to use. I haven´t checked the performance burden of it